core/src/components/toast/toast.tsx

```diff
@@ -281,6 +287,10 @@ export class Toast implements ComponentInterface, OverlayInterface {
           <div class="toast-container" part="container">
             {this.renderButtons(startButtons, 'start')}
 
+            {this.icon !== undefined &&
+              <ion-icon class="toast-icon" part="icon" icon={this.icon}></ion-icon>
```

Comment:
```suggestion
              <ion-icon class="toast-icon" part="icon" icon={icon} lazy={false} aria-hidden="true"></ion-icon>
```

The icon does not serve a functional purpose, so I don't think we need to announce it to screen readers. We can add `lazy` here so that the icon is loaded when the `ion-toast` is first created, not when the toast comes into view.

Let's also add `icon` to the first line in the `render` function so that we don't need to keep accessing `this`.